src/webgpu/api/validation/createSampler.spec.ts

```diff
@@ -0,0 +1,43 @@
+export const description = `
+createSampler validation tests.
+`;
+
+import { poptions, params } from '../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+
+import { ValidationTest } from './validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+g.test('lodMinAndMaxClamp')
+  .desc('test different combinations of min and max clamp values')
+  .params(
+    params()
+      .combine(poptions('lodMinClamp', [-4e-30, -1, 0, 0.5, 1, 10, 4e30]))
+      .combine(poptions('lodMaxClamp', [-4e-30, -1, 0, 0.5, 1, 10, 4e30]))
+  )
+  .fn(async t => {
+    if (t.params.lodMinClamp > t.params.lodMaxClamp) {
+      t.expectValidationError(() => {
+        t.device.createSampler({
+          lodMinClamp: t.params.lodMinClamp,
+          lodMaxClamp: t.params.lodMaxClamp,
+        });
+      });
+    } else if (
+      t.params.lodMinClamp < 0 ||
+      t.params.lodMaxClamp < 0
+    ) {
+      t.expectValidationError(() => {
+        t.device.createSampler({
+          lodMinClamp: t.params.lodMinClamp,
+          lodMaxClamp: t.params.lodMaxClamp,
+        });
+      });
+    } else {
+      t.device.createSampler({
+        lodMinClamp: t.params.lodMinClamp,
+        lodMaxClamp: t.params.lodMaxClamp,
+      });
+    }
```

Comment:
expectValidationError can take a boolean for whether to expect a validation error or not. Can you compute that boolean using similar logic, and then just have one createSampler call?